### DISCUSSION TITLE
With Apollo server, how can I use `info` with prisma-client

### DISCUSSION THREAD

**Discussion by alan345** [prisma/prisma#2910] [Q&A] [answered]
With Apollo server, prisma1 I have a resolver:
```ts
  async invoicesConnection(parent: any, args: any, ctx: any, info: any) {
    const invoices = await ctx.db.query.invoicesConnection(args, info)
    return invoices
  },
```

Now, how can I do it with prisma2, prisma-client and Apollo server?
```ts
    const invoices = prisma.invoices.findMany({
      skip: args.skip,
      take: args.first,
      where: args.where
    })
```
What about `info` to select only the fields required by the frontend?

**Comment by ryands17** [comment, marked as answer]
Hey @alan345 :wave: 
For Prisma 2, you would have to manually extract fields from `info` and then pass into `select` in Prisma like this:

```ts
const fields = 'these are the fields you collected from info'
const invoices = prisma.invoices.findMany({
      skip: args.skip,
      take: args.first,
      where: args.where,
     select: fields
})
```

Or you can use a package that does the same: https://paljs.com/plugins/select/

**Comment by alan345** [comment]
Hey @ryands17 
Can you give me more details about `const fields = 'these are the fields you collected from info'`

In this example: 
```ts
    const invoices = await ctx.prisma.invoice.findMany({
      where: args.where,
      include: {
        user: true, // optional based on `info`
      },
      skip: args.skip,
      take: args.first,
    })
```

I want to include `user` only if the frontend is requesting it. The data is supposed to be in `info`, but where?

**Reply by ryands17** [reply]
Hey @alan345 
In your resolver, you will be able to find the fields here:

```ts
resolve(_parent, _args, ctx, info) {
  // this will contain the fields requested
  console.log(JSON.stringify(info.fieldNodes, null, 2))

  const invoices = await ctx.prisma.invoice.findMany({
      where: args.where,
      include: {
        user: true, // optional based on `info`
      },
      skip: args.skip,
      take: args.first,
    })
},
```

From here, you can write a transform function that gets all the field names from this object.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
